Add org-hyperscheduler to melpa

dmitrym0 opened the pull request:
### Brief summary of what the package does

org-hyperscheduler provides a web based view of your org-agenda.

### Direct link to the package repository

https://github.com/dmitrym0/org-hyperscheduler


### Your association with the package

I'm the author of org-hyperscheduler.

### Checklist

- [x] The package is released under a [GPL-Compatible Free Software License](https://www.gnu.org/licenses/license-list.en.html#GPLCompatibleLicenses)
- [x] I've read [CONTRIBUTING.org](https://github.com/melpa/melpa/blob/master/CONTRIBUTING.org)
- [x] I've used the latest version of [package-lint](https://github.com/purcell/package-lint) to check for packaging issues, and addressed its feedback
- [x] My elisp byte-compiles cleanly
- [x] `M-x checkdoc` is happy with my docstrings
- [x] I've built and installed the package using the instructions in [CONTRIBUTING.org](https://github.com/melpa/melpa/blob/master/CONTRIBUTING.org)
- [nice] I have confirmed some of these without doing them

riscy commented:
Thanks for your patience!  Quick first pass --

`org-hyperscheduler.el` with byte-compile using Emacs 28.1:
```
org-hyperscheduler.el:52:1: Warning: custom-declare-variable
    `org-hyperscheduler-readonly-mode' docstring wider than 80 characters
org-hyperscheduler.el:64:1: Warning: custom-declare-variable
    `org-hyperscheduler-hide-done-tasks' docstring wider than 80 characters
org-hyperscheduler.el:81:1: Warning: custom-declare-variable
    `org-hyperscheduler-agenda-filter' docstring wider than 80 characters
In org-hyperscheduler-get-calendar-entries:
org-hyperscheduler.el:213:8: Warning: docstring wider than 80 characters
```
These are minor, but easy to fix I think.

Package and license:
- Prefer the default recipe or `:defaults`, if possible.  MELPA doesn't usually distribute the LICENSE file, so I would remove that from the recipe too
- Some of your non-elisp code, like index.js, doesn't contain license boilerplate that users would be able to see
- The code seems to contain some print debugging, e.g. `(message "+org-hyperscheduler-update-event")` and `(message "-org-hyperscheduler-update-event")` when entering/leaving a function; ideally you remove these so users only see the messages they need to see
- In `org-hyperscheduler--ws-on-open`, the `progn` looks unnecessary

riscy commented:
I'll mark this awaiting-upstream for now, but feel free to ping for another look.

dmitrym0 commented:
@riscy, appreciate your feedback:

- reformatted the doc strings
- I removed the LICENSE file from the recipe, however I'm not sure how `:defaults` can handle the `calendar/` subdirectory.  I can flatten the directory structure, but then I have to mix `.js` and `.el` files in my source tree and that doesn't seem that nice. What are your thoughts?
- Added standard GPL header to index.js
- Replaced `(message)` with log4e logging; most output will be supressed.
- Removed `progn`.

riscy commented on recipes/org-hyperscheduler:
```diff
@@ -0,0 +1,4 @@
+(org-hyperscheduler :repo "dmitrym0/org-hyperscheduler"
+                    :fetcher github
+                    :files ("org-hyperscheduler.el" "LICENSE"
+                           ("calendar" "calendar/index.html" "calendar/index.js" "calendar/org-hs.css")))
```
> I removed the LICENSE file from the recipe ...

Here's how I'd apply `:defaults` in this case, but check my work:

```suggestion
(org-hyperscheduler :repo "dmitrym0/org-hyperscheduler"
                    :fetcher github
                    :files (:defaults
                            ("calendar"
                             "calendar/index.html"
                             "calendar/index.js"
                             "calendar/org-hs.css")))
```

riscy reviewed (commented):
With the recent code changes there are a few more nags, but these should be easy fixes -

`org-hyperscheduler.el` with checkdoc 0.6.2:
```
org-hyperscheduler.el:69: First line is not a complete sentence
org-hyperscheduler.el:87: First line is not a complete sentence
org-hyperscheduler.el:182: First sentence should end with punctuation
org-hyperscheduler.el:233: First line is not a complete sentence
```

- org-hyperscheduler.el#L102: Single-line comments should usually begin with `;;`
- org-hyperscheduler.el#L178: Prefer one blank line between this top-level form and the next
- org-hyperscheduler.el#L254: You can use `not` or `null`
- org-hyperscheduler.el#L101: Top-level `setq` should be replaced by `defvar` (or, in this case, it looks like you're modifying a variable that shouldn't necessarily be modified at load time)

dmitrym0 commented:
@riscy thanks for the feedback. I did miss running check doc somehow.

1. Resolved all check doc issues.
2. Removed extra blank lines between `defun`s.
3. Updates comments to `;;`.
4. Removed the `setq`.
5. Finally, updated the recipe to use `:defaults`.

Thanks for taking the time to look at this!

riscy commented:
Thanks for your patience!  This is excellent. :)
